src/go/wsl-helper/pkg/dockerproxy/serve.go

```diff
@@ -48,32 +64,313 @@ func serve(endpoint string, dialer func() (net.Conn, error)) error {
+// arbitrary mapping that can be reused in the response munger, as well as a
+// mapping of any path templating patterns that were matched.
+type requestMungerFunc func(*http.Request, *RequestContextValue, map[string]string) error
+
+// responseMungerFunc is a munger for an outgoing response; it also receives an
+// arbitrary mapping that was initially passed to the matching request munger,
+// as well a a mapping of any path templating patterns that were matched.
+type responseMungerFunc func(*http.Response, *RequestContextValue, map[string]string) error
+
+// mungerMethodMapping is a helper structure to find a munger given an API path,
+// specialized for a given HTTP method (GET, POST, etc.).
+// This should only be written to during init(), at which point it's protected
+// by the lock on mungerMapping.
+type mungerMethodMapping struct {
+	// requests that are simple (have no path templating)
+	requests map[string]requestMungerFunc
+	// requestPatterns are requests that involve path templating
+	requestPatterns map[*regexp.Regexp]requestMungerFunc
+	// responses that are simple (have no path templating)
+	responses map[string]responseMungerFunc
+	// responsePatterns are responses that involve path templating
+	responsePatterns map[*regexp.Regexp]responseMungerFunc
+}
+
+// getRequestMunger gets the munger to use for this request, as well as the
+// path templating elements (if relevant for the munger).
+func (m *mungerMethodMapping) getRequestMunger(apiPath string) (requestMungerFunc, map[string]string) {
+	if munger, ok := m.requests[apiPath]; ok {
+		return munger, nil
+	}
+	for pattern, munger := range m.requestPatterns {
+		matches := pattern.FindStringSubmatch(apiPath)
+		if matches != nil {
+			names := pattern.SubexpNames()
+			results := make(map[string]string)
+			for i, name := range names {
+				results[name] = matches[i]
+			}
+			return munger, results
+		}
+	}
+	return nil, nil
+}
+
+func (m *mungerMethodMapping) getResponseMunger(apiPath string) (responseMungerFunc, map[string]string) {
+	if munger, ok := m.responses[apiPath]; ok {
+		return munger, nil
+	}
+	for pattern, munger := range m.responsePatterns {
+		matches := pattern.FindStringSubmatch(apiPath)
+		if matches != nil {
+			names := pattern.SubexpNames()
+			results := make(map[string]string)
+			for i, name := range names {
+				results[name] = matches[i]
+			}
+			return munger, results
+		}
+	}
+	return nil, nil
+}
+
+// mungerMapping contains mungers that will handle particular API endpoints.
+var mungerMapping struct {
+	sync.RWMutex
+	mungers map[string]*mungerMethodMapping
+}
+
+// convertPattern converts an API path to a regular expression pattern for
+// matching URLs with path templating; if there are no path templates, this
+// returns nil.  The returned pattern always matches the whole string.
+func convertPattern(apiPath string) *regexp.Regexp {
+	matches := regexp.MustCompile(`\{[^}/]+\}`).FindAllStringIndex(apiPath, -1)
```

Comment:
I don't think you need to escape the escaped braces -- the `{` because it can't denote a repetition when at the start of a pattern, and the second `}` because you aren't in a repetition specifier.

Reply:
Fixed in 396638ad550e052afd8d5960a71ca27e0269ed6.